src/components/sections/events/EventsClient.tsx

```diff
@@ -22,13 +22,19 @@ export default function EventsClient({
   const [locationFilter, setLocationFilter] = useState<string | null>(null);
 
   const allLocations = Array.from(
-    new Set(eventPostings.flatMap((event) => event.locations)),
+    new Set(
+      eventPostings.flatMap(
+        (event) => event.locations?.map((loc) => loc.locationObject) ?? [],
```

Comment:
Veldig mye undefined grokking i forskjellige linjer, ikke kritisk å fikse opp i nå men generelt (og spesielt med lister) er det fint å prøve å defaulte til verdier så tidlig som mulig for å unngå å måtte propagere undefined/null igjennom hele kodebasen. For lister så er det jo ganske enkelt i all den grad en faktisk ikke trenger å vite om den er undefined. For da kan en bruke en tom liste og behandle den som forventet enten den er tom eller full.

Reply:
Jeg ble så stressa av at jeg fikk feilmeldinger over alt på grunn av `undefined`, så jeg bare fylte på med checker for å være sikker på at det ikke skulle skje, hehe.